dev-libs/libsecp256k1/libsecp256k1-0.1_pre20211216.ebuild

```diff
@@ -0,0 +1,106 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit autotools flag-o-matic toolchain-funcs
+
+MyPN=secp256k1
+DESCRIPTION="Optimized C library for EC operations on curve secp256k1"
+HOMEPAGE="https://github.com/bitcoin-core/secp256k1"
+COMMITHASH="0559fc6e41b65af6e52c32eb9b1286494412a162"
+SRC_URI="https://github.com/bitcoin-core/${MyPN}/archive/${COMMITHASH}.tar.gz -> ${PN}-v${PV}.tgz"
+
+LICENSE="MIT"
+SLOT="0/20210628"  # subslot is date of last ABI change
+KEYWORDS="~amd64 ~arm ~arm64 ~mips ~ppc ~ppc64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="+asm ecdh +experimental +extrakeys lowmem precompute-ecmult +schnorr +recovery test valgrind"
+RESTRICT="!test? ( test )"
+
+REQUIRED_USE="
+	asm? ( || ( amd64 arm ) arm? ( experimental ) )
+	extrakeys? ( experimental )
+	?? ( lowmem precompute-ecmult )
+	schnorr? ( experimental extrakeys )
+"
+RDEPEND="
+	!=dev-util/bitcoin-tx-0.21* !=dev-util/bitcoin-tx-21.2-r0
```

Comment:
This is likely to just lead to confusing errors for people. I would instead revbump all existing versions in tree (`git mv` them all) to depend on < this version.

Reply:
Won't that allow upgrading libsecp256k1, and leave their current version broken?

Reply:
Sorry, what I was saying was, we should use pure < deps (which would not allow their current version to be broken) rather than blockers which are not strictly for this (they're purely for conflicting files, really).

The effect is the same but the method of achieving it is different, with some nicer emerge output.

Reply:
Summary of IRC discussion: Using < deps won't work because there's bumps to the middle versions and <0.21 are also fine. So this is the best we can do here.